.github/workflows/release.yml

```diff
@@ -0,0 +1,111 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      dryRun:
+        description: 'Do a dry run to preview instead of a real release'
+        required: true
+        default: 'true'
+
+jobs:
+  authorize:
+    name: Authorize
+    runs-on: macos-10.14
+    steps:
+      - name: ${{ github.actor }} permission check to do a release
+        uses: octokit/request-action@v2.0.0
+        with:
+          route: GET /repos/:repository/collaborators/${{ github.actor }}
+          repository: ${{ github.repository }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  release:
+    name: Release
+    runs-on: macos-10.14
+    needs: [authorize]
+    strategy:
+      matrix:
+        ruby-version: ["2.7.x"]
+        node-version: ["12.x"]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    
+    - name: Set Xcode 12
+      run: |
+        sudo xcode-select -switch /Applications/Xcode_12.app
+
+    - name: Setup Ruby
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby-version }}
+
+    # - name: Cache Bundle Gems and Cocoapods
+    #   id: cache-gems-pods
+    #   uses: actions/cache@v2
+    #   with:
+    #     path: |
+    #       Pods
+    #       vendor/bundle
+    #     key: ${{ runner.os }}-ruby-${{ matrix.ruby-version }}-gems-${{ hashFiles('**/Gemfile.lock') }}-pods-${{ hashFiles('**/Podfile.lock') }}
+    
+    - name: Install Cocoapods
+      # if: steps.cache-gems-pods.outputs.cache-hit == 'false'
+      run: |
+        gem install bundler
+        bundle config path vendor/bundle
+        bundle install
+        pod install
+
+    - name: Run tests
+      run: |
+        xcodebuild test \
+          -workspace Amplitude.xcworkspace \
+          -scheme Amplitude_iOSTests \
+          -sdk iphonesimulator \
+          -destination 'platform=iOS Simulator,name=iPhone 11,OS=14.0'
+    
+    - name: Validate Podfile
+      run: |
+        pod spec lint
+
+    - name: Setup Node.js
```

Comment:
Do you need this?

Reply:
Needed for running [`semantic-release`](https://github.com/amplitude/Amplitude-iOS/pull/291/files/545c7ff0f10b3bd310b2366cf941c59aa8051c35#diff-56c30d9224897b08c28075052d4595c5R79) since it's a Node based tool.